File: frontend/src/core/services/market.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { BaseResponse } from '../models/auth.model';
import { environment } from '../../environments/environment';
import {
    MarketSummaryResponse,
    MarketTopGainerResponse,
    MarketTopLiquidityResponse,
    MarketSectorResponse
} from '../models/market.model';

@Injectable({ providedIn: 'root' })
export class MarketService {
    private readonly http    = inject(HttpClient);
    private readonly API_URL = `${environment.apiUrl}/api/market`;

    getSummary(): Observable<MarketSummaryResponse> {
        return this.http.get<BaseResponse<MarketSummaryResponse>>(`${this.API_URL}/summary`).pipe(
            map(r => r.data)
        );
    }

    getTopGainers(): Observable<MarketTopGainerResponse[]> {
        return this.http.get<BaseResponse<MarketTopGainerResponse[]>>(`${this.API_URL}/top-gainers`).pipe(
            map(r => r.data)
        );
    }

    getTopLiquidity(): Observable<MarketTopLiquidityResponse[]> {
        return this.http.get<BaseResponse<MarketTopLiquidityResponse[]>>(`${this.API_URL}/top-liquidity`).pipe(
            map(r => r.data)
        );
    }

    getSectors(): Observable<MarketSectorResponse[]> {
        return this.http.get<BaseResponse<MarketSectorResponse[]>>(`${this.API_URL}/sectors`).pipe(
            map(r => r.data)
        );
    }
}

File: app/src/main/java/com/hieunt/stock/service/MarketService.java
package com.hieunt.stock.service;

import java.util.List;

import com.hieunt.stock.model.response.MarketSectorResponse;
import com.hieunt.stock.model.response.MarketSummaryResponse;
import com.hieunt.stock.model.response.MarketTopGainerResponse;
import com.hieunt.stock.model.response.MarketTopLiquidityResponse;

public interface MarketService {
    MarketSummaryResponse getSummary();

    List<MarketTopGainerResponse> getTopGainers();

    List<MarketTopLiquidityResponse> getTopLiquidity();

    List<MarketSectorResponse> getSectors();
}
